lib/ProductOpener/Packaging.pm

```diff
@@ -846,6 +851,71 @@ sub set_packaging_misc_tags ($product_ref) {
 	return;
 }
 
+=head2 aggregate_packaging_by_parent_materials ($product_ref)
+
+Aggregate the weights of each packaging component by parent material (glass, plastics, metal, paper or cardboard)
+
+=cut
+
+sub aggregate_packaging_by_parent_materials ($product_ref) {
+
+	my $packagings_materials_ref = {"all" => {}};
+
+	if (defined $product_ref->{packagings}) {
+
+		# Iterate over each packaging component
+		foreach my $packaging_ref (@{$product_ref->{packagings}}) {
+
+			# Determine what is the parent material for the component
+			my $material = $packaging_ref->{material};
+			my $parent_material = "en:unknown";
+			if (defined $material) {
+				foreach my $parent ("en:paper-or-cardboard", "en:plastic", "en:glass", "en:metal") {
+					if (is_a("packaging_materials", $material, $parent)) {
+						$parent_material = $parent;
+						last;
+					}
+				}
+			}
```

Comment:
I don't know enough of the source code to suggest an entirely different approach to the logic itself, but a common pattern for getting the first item in a list that matches a certain criteria is using [List::Util's `first`](https://metacpan.org/pod/List::Util#first):

```perl
use List::Util qw(first);

$parent_material = first {
      is_a("packaging_materials", $material, $_)
} ("en:paper-or-cardboard", "en:plastic", "en:glass", "en:metal");
``` 

but if this code is hot (i.e. running several times per call) maybe it could be worth turning `is_a()` into a hash lookup.

Either way, the current implementation is okay and I wouldn't change it unless it was a bottleneck.

Reply:
Thanks for the suggestions. I changed to @garu 's suggestion as I find it more readable with first indeed.

It certainly could be done more efficiently, but this code is executed only when writing products, and given the shallowness of the packaging materials taxonomy, it's just a dozen operations, so I'd prefer not to spend time on it. If we want to optimize things, there are targets that are much more likely to make a difference (e.g. for read operations of products, where we compute attributes and knowledge panels for 100 products...)